### PR TITLE
Added notebook demonstrating hyperparameter tuning

### PR DESCRIPTION
**Pull request by nikitamaia**
Created notebook demonstrating hyperparameter tuning for a TensorFlow model

### CODE REVIEW

File: notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb [hunk 1]
```diff
@@ -0,0 +1,1225 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "58f02d71787a"
+      },
+      "source": [
+        "Next, define the `parameter_spec`, which is a dictionary specifying the parameters you want to optimize. The **dictionary key** is the string you assigned to the command line argument for each hyperparameter, and the **dictionary value** is the parameter specification.\n",
+        "\n",
+        "For each hyperparameter, you need to define the Type as well as the bounds for the values that the tuning service will try. Hyperparameters can be of type Double, Integer, Categorical, or Discrete. If you select the type Double or Integer, you'll need to provide a minimum and maximum value. And if you select Categorical or Discrete you'll need to provide the values. For the Double and Integer types, you'll also need to provide the Scaling value. You can learn more about how to pick the best scale [in this video](https://www.youtube.com/watch?v=cSoK_6Rkbfg&list=PLkDaE6sCZn6Hn0vK8co82zjQtt3T2Nkqc&index=25)."
```
Comment (jackiejackjack) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1028]: Variable `Type` and all it's possible values should be in code font.

File: notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb [hunk 2]
```diff
@@ -0,0 +1,1225 @@
...
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "c82fd780f019"
+      },
+      "source": [
+        "Then, create and run the HyperparameterTuningJob."
```
Comment (jackiejackjack) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1103]: Then, create and run `HyperparameterTuningJob`.

File: notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb [hunk 3]
```diff
@@ -0,0 +1,1225 @@
...
+        "\n",
+        "* **parallel_trial_count**: If you use parallel trials, the service provisions multiple training processing clusters. Increasing the number of parallel trials reduces the amount of time the hyperparameter tuning job takes to run; however, it can reduce the effectiveness of the job overall. This is because the default tuning strategy uses results of previous trials to inform the assignment of values in subsequent trials.\n",
+        "\n",
+        "* **search_algorithm**: You can set the search algorithm to grid, random, or default (None). The default option applies Bayesian optimization to search the space of possible hyperparameter values and is the recommended algorithm. You can learn more about this algorithm here."
```
Comment (jackiejackjack) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1139]: Is "none" the default or is "default" none? Or does "none" refer to leaving this configuration blank?

**Review by jackiejackjack — APPROVED** [review]

File: notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb [hunk 4]
```diff
@@ -0,0 +1,1229 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1]: <p>you'll -&gt; you</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1168/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (andrewferlitsch) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1]: <p>Add heading to text cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1168/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (andrewferlitsch) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1]: <p>combine the 2 text cells</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1168/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (andrewferlitsch) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1]: <p>you'll -&gt; you</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1168/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (andrewferlitsch) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1168/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

Comment (andrewferlitsch) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1]: <p>combine 2 code cells</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1168/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (andrewferlitsch) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1]: <p>combine code cells</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1168/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='50'/>

Comment (andrewferlitsch) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1]: <p>Lastly -&gt; Finally</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1168/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (andrewferlitsch) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1]: <p>combine the 3 text cells</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1168/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

Comment (andrewferlitsch) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1]: <p>Add heading</p><p>you'll -&gt; you</p><p> </p><p>For links, use pattern</p><p> </p><p>Learn more about [title](link)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1168/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='59'/>

Comment (andrewferlitsch) [inline notebooks/official/training/hyperparameter_tuning_tensorflow.ipynb:1]: <p>combine 2 text cells</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1168/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='68'/>

**Review by andrewferlitsch — APPROVED** [review]

**Pull request merged**